remediation/workflow/pin/pinactions.go

```diff
@@ -76,6 +76,13 @@ func PinAction(action, inputYaml string) (string, bool) {
 	pinnedAction := fmt.Sprintf("%s@%s # %s", leftOfAt[0], commitSHA, tagOrBranch)
 	updated = !strings.EqualFold(action, pinnedAction)
 	inputYaml = strings.ReplaceAll(inputYaml, action, pinnedAction)
+	stringParts := strings.SplitN(inputYaml, pinnedAction, 2)
```

Comment:
@Devils-Knight can you please review this code? I think you had worked on this code recently, so just review this logic once. Also confirm if this change is on the latest from `main`.